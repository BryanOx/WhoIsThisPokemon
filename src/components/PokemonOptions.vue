<template>
  <div class="options-container">
    <ul>
      <li
        @click="$emit('selection', pokemon.id)"
        :key="pokemon.id"
        v-for="pokemon in pokemons"
        :class="
          pokemon.id === correctAnswer.id && correctAnswer.correct
            ? 'correct'
            : !correctAnswer.correct && correctAnswer.id === pokemon.id
            ? 'incorrect'
            : 'null'
        "
      >
        {{ capitalize(pokemon.name) }}
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "@vue/runtime-core";

export default defineComponent({
  props: {
    pokemons: {
      type: Array,
      required: true
    },
    correctAnswer: Object
  },
  methods: {
    capitalize(name: string): string {
      return name.toUpperCase();
    }
  }
});
</script>

<style scoped>
ul {
  list-style-type: none;
}

li {
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

li.null {
  background-color: white;
}

li.correct {
  background-color: rgb(24, 181, 24) !important;
  color: white;
  font-weight: 900;
}

li.incorrect {
  background-color: rgb(201, 31, 31) !important;
  color: white;
  font-weight: 900;
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
