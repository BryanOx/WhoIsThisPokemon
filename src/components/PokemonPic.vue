/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
<template>
  <div data-testid="pokemonContainer" class="pokemon-container">
    <img
      class="hidden-pokemon"
      data-testid="pokemonHidden"
      :src="imgSrc"
      v-if="!showPokemon"
      alt="A Wild  Pokemon"
    />
    <img
      class="fade-in"
      data-testid="pokemonUnhidden"
      :src="imgSrc"
      v-else
      alt="A Wild  Pokemon"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent } from "@vue/runtime-core";

export default defineComponent({
  name: 'PokemonPic',
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    }
  },
  computed: {
    imgSrc(): string {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    }
  },
});
</script>

<style scope>
.pokemon-container {
  height: 200px;
}
img {
  height: 200px;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.hidden-pokemon {
  filter: brightness(0);
}
</style>
