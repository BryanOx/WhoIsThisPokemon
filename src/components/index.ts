export { default as PokemonOptions } from './PokemonOptions.vue';
export { default as PokemonPic } from './PokemonPic.vue';
