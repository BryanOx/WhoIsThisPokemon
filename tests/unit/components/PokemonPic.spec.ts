/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallowMount, VueWrapper } from "@vue/test-utils";

import { PokemonPic } from "@/components/";

describe("PokemonPic <component>", () => {
  function mountWrapper(id = 1, showPokemon = false): VueWrapper<any> {
    return shallowMount(PokemonPic, {
      props: {
        pokemonId: id,
        showPokemon: showPokemon
      }
    });
  }

  it("Should match with snapshot", () => {
    const wrapper = mountWrapper();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should show the hidden image of the pokemon 100", () => {
    const wrapper = mountWrapper(100, false);
    const img = wrapper.find("img");
    expect(img.exists()).toBeTruthy();
    expect(img.classes("hidden-pokemon")).toBeTruthy();
    expect(img.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
    expect(
      wrapper.find('[data-testid="pokemonUnhidden"]').exists()
    ).toBeFalsy();
    expect(wrapper.props().pokemonId).toBe(100);
  });

  it("Should pokemon if showPokemon is true", () => {
    const wrapper = mountWrapper(100, true);
    const img = wrapper.find("img");
    expect(img.exists()).toBeTruthy();
    expect(img.classes("fade-in")).toBeTruthy();
    expect(img.attributes("src")).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg"
    );
    expect(wrapper.props().showPokemon).toBeTruthy();
  });
});
