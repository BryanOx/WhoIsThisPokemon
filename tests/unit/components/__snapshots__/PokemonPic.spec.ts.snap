// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPic <component> Should match with snapshot 1`] = `<div data-testid="pokemonContainer" class="pokemon-container"><img class="hidden-pokemon" data-testid="pokemonHidden" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="A Wild  Pokemon"></div>`;
